src/Build.UnitTests/BackEnd/BuildRequestConfiguration_Tests.cs

```diff
@@ -260,6 +260,51 @@ public void TestTranslation()
             Assert.Equal(config, deserializedConfig);
         }
 
+        [Fact]
+        public void TestTranslationWithEntireProjectState()
```

Comment:
Do you think it would make sense to test all combinations of

`TranslateEntireState` was set/unset via escape hatch on the write side
x
`TranslateEntireState` was set/unset via escape hatch on the read side

This would cover what's described in [this your comment](https://github.com/dotnet/msbuild/pull/6385#discussion_r625072838).